lib/spack/spack/binary_distribution.py

```diff
@@ -631,3 +708,114 @@ def get_keys(install=False, trust=False, force=False):
                 else:
                     tty.msg('Will not add this key to trusted keys.'
                             'Use -t to install all downloaded keys')
+
+
+def read_from_url(file_uri):
+    file_contents = None
+
+    try:
+        url = urlopen(file_uri)
+    except URLError as url_err:
+        msg = 'Unable to open url {0} due to {1}'.format(
+            file_uri, url_err.message)
+        raise spack.error.SpackError(msg)
```

Comment:
probably ok not to wrap this here, as the function's only used by routines in this file -- the SpackError is immediately caught by `needs_rebuild`, anyway -- you could just let it catch the `URLError`.